spec/rubocop/cop/style/space_around_operators_spec.rb

```diff
@@ -276,55 +162,362 @@ def check_modifier(keyword)
+                           'y !~/abc/'])
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `=~`.",
+                "Surrounding space missing for operator `!~`."])
+    end
+
+    it 'registers an offense for various assignments without space' do
+      inspect_source(cop, ['x||=0', 'y&&=0', 'z*=2',
+                           '@a=0', '@@a=0', 'a,b=0', 'A=0', 'x[3]=0', '$A=0'])
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `||=`.",
+                "Surrounding space missing for operator `&&=`.",
+                "Surrounding space missing for operator `*=`.",
+                "Surrounding space missing for operator `=`.",
+                "Surrounding space missing for operator `=`.",
+                "Surrounding space missing for operator `=`.",
+                "Surrounding space missing for operator `=`.",
+                "Surrounding space missing for operator `=`.",
+                "Surrounding space missing for operator `=`."])
+    end
+
+    it 'registers an offense for equality operators without space' do
+      inspect_source(cop, ['x==0', 'y!=0', 'Hash===z'])
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `==`.",
+                "Surrounding space missing for operator `!=`.",
+                "Surrounding space missing for operator `===`."])
+    end
+
+    it 'registers an offense for - without space with negative lhs operand' do
+      inspect_source(cop, '-1-arg')
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `-`."])
+    end
+
+    it 'registers an offense for inheritance < without space' do
+      inspect_source(cop, ['class ShowSourceTestClass<ShowSourceTestSuperClass',
+                           'end'])
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `<`."])
+    end
+
+    it 'registers an offense for hash rocket without space at rescue' do
+      inspect_source(cop, ['begin',
+                           'rescue Exception=>e',
+                           'end'])
+      expect(cop.messages)
+        .to eq(["Surrounding space missing for operator `=>`."])
+    end
+  end
+
+  describe 'extra space around operators' do
+    before do
+      # Don't accept multiple spaces around `=` in these tests.
+      cop_config['MultiSpaceAllowedForOperators'] = []
+    end
+
+    it 'registers an offense for assignment with many spaces on either side' do
+      inspect_source(cop, ['x  = 0', 'y +=  0', 'z[0]  =  0'])
+      expect(cop.messages)
+        .to eq(["Operator `=` should be surrounded with a single space.",
+                "Operator `+=` should be surrounded with a single space.",
+                "Operator `=` should be surrounded with a single space."])
+    end
+
+    it 'accepts assignment with many spaces on either side, if so configured' do
+      cop_config['MultiSpaceAllowedForOperators'] = ['=']
```

Comment:
@bbatsov I am not really sure about this spec… I’ve chosen to disable the default configuration in line 338, and only in this single spec do I check that the configuration actually works.

Should I rather leave the default configuration and add specs for *that* behavior – and then add a single spec showing that adding a `MultiSpaceAllowedForOperators` setting changes the cop’s behavior?